kogito-codegen-modules/kogito-codegen-rules/src/main/java/org/kie/kogito/codegen/rules/IncrementalRuleCodegen.java

```diff
@@ -171,18 +172,16 @@ public boolean isEmpty() {
             throw new RuleCodegenError(modelBuilder.getErrors().getErrors());
         }
 
-        Map<String, String> modelsByUnit = new HashMap<>();
-
-        List<GeneratedFile> generatedFiles = new ArrayList<>(generateModels(modelBuilder, modelsByUnit));
-
-        boolean hasRuleUnits = !ruleUnitGenerators.isEmpty();
+        Map<String, String> modelsByPackage = new HashMap<>();
```

Comment:
Considering that this is only for legacy, please rename

```suggestion
        Map<String, String> modelsByPackageForLegacyApi = new HashMap<>();
```